client/app/components/ParameterValueInput.jsx

```diff
@@ -128,31 +131,41 @@ export class ParameterValueInput extends React.Component {
   }
 
   renderEnumInput() {
-    const { value, onSelect, enumOptions } = this.props;
+    const { value, onSelect, enumOptions, allowMultipleValues } = this.props;
     const enumOptionsArray = enumOptions.split('\n').filter(v => v !== '');
     return (
       <Select
-        className={this.props.className}
+        className={classNames({ 'parameter-multi-select': allowMultipleValues }, this.props.className)}
+        mode={allowMultipleValues ? 'multiple' : 'default'}
+        optionFilterProp="children"
         disabled={enumOptionsArray.length === 0}
-        defaultValue={value}
+        value={value}
```

Comment:
Otherwise the changes from `setValue` wouldn't affect here.

Other inputs remain the same as is, one need is to see the behavior after #3907.